src/Resources/ResourceManager/Utilities/WhatIfResourceIdParser.cs

```diff
@@ -0,0 +1,70 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Utilities
+{
+    using System.Text.RegularExpressions;
+
+    public static class WhatIfResourceIdParser
```

Comment:
Any reason why this is a *WhatIf*ResourceIdParser rather than a generic one?

Reply:
Good point. Initially I was thinking that it will only be used by what-if models, and what-if only support subscription-level resource IDs. But we may in the future add support for MG and tenant level resources, so making it generic makes sense.